### PR TITLE
Use annotations to configure rules in detekt-rules-exceptions

### PR DESCRIPTION
**Pull request by marschwar**
This belongs to #3670 and replaces all configuration kdoc tags with annotations.

### CODE REVIEW

File: detekt-core/src/main/resources/default-detekt-config.yml [hunk 1]
```diff
@@ -203,11 +203,7 @@ exceptions:
     ignoreLabeled: false
   SwallowedException:
     active: true
-    ignoredExceptionTypes:
-     - InterruptedException
-     - NumberFormatException
-     - ParseException
-     - MalformedURLException
+    ignoredExceptionTypes: ['NumberFormatException', 'InterruptedException', 'ParseException', 'MalformedURLException']
```
Comment (marschwar) [inline detekt-core/src/main/resources/default-detekt-config.yml:206]: I realize this is a problem for git diffs. It is already adressed in #3671

File: detekt-core/src/main/resources/default-detekt-config.yml [hunk 2]
```diff
@@ -216,32 +212,17 @@ exceptions:
   ThrowingExceptionsWithoutMessageOrCause:
     active: true
     excludes: ['**/test/**', '**/androidTest/**', '**/commonTest/**', '**/jvmTest/**', '**/jsTest/**', '**/iosTest/**']
-    exceptions:
-     - IllegalArgumentException
-     - IllegalStateException
-     - IOException
+    exceptions: ['ArrayIndexOutOfBoundsException', 'Error', 'Exception', 'IllegalMonitorStateException', 'NullPointerException', 'IndexOutOfBoundsException', 'RuntimeException', 'Throwable']
```
Comment (marschwar) [inline detekt-core/src/main/resources/default-detekt-config.yml:215]: There previously was a disconnect between the documented and the actual default. Which one should we keep?

**Review by schalkms — APPROVED** [review]

File: detekt-core/src/main/resources/default-detekt-config.yml (hunk 2 above)
Comment (BraisGabin) [reply detekt-core/src/main/resources/default-detekt-config.yml:215]: I think that these list is better.

File: detekt-core/src/main/resources/default-detekt-config.yml (hunk 1 above)
Comment (BraisGabin) [reply detekt-core/src/main/resources/default-detekt-config.yml:206]: I agree with @3flex in that issue, use multiline in all of them.

**Review by BraisGabin — APPROVED** [review]

File: detekt-core/src/main/resources/default-detekt-config.yml (hunk 2 above)
Comment (cortinico) [reply detekt-core/src/main/resources/default-detekt-config.yml:215]: > There previously was a disconnect between the documented and the actual default.

As a rule of thumb: `code >> documentation`. Always prefer what the code says as docs could get out of date

File: detekt-core/src/main/resources/default-detekt-config.yml (hunk 1 above)
Comment (cortinico) [reply detekt-core/src/main/resources/default-detekt-config.yml:206]: Are you following up to multiline in a separate PR?

**Review by cortinico — APPROVED** [review]

File: detekt-core/src/main/resources/default-detekt-config.yml (hunk 1 above)
Comment (marschwar) [reply detekt-core/src/main/resources/default-detekt-config.yml:206]: Yes I am working on a fix for the mutliline issue.

**Pull request merged**